bash-completion: refactor to python extension

ignatenkobrain opened the pull request:
@jsilhan @rholy We can make the same api as we did for `dnf.command.Command`.

jsilhan commented on etc/bash_completion.d/dnf-completion.bash:
```diff
@@ -71,106 +74,39 @@ _dnf()
+            install|update|upgrade)
+                if [[ "$cur" != \.* ]] && [[ "$cur" != \/* ]] && [[ "$cur" != ~* ]]; then
+                    _dnf_helper $command "$cur"
+                    # FIXME: fix sqlitedb to store hawkey.Package
```
How you wanna store hawkey.Package object? I doubt it would work even with pickle when the main part is in C Python.

jsilhan commented:
@ignatenkobrain, whats the motivation of this change? Before that we had separated code for completion and now we have one more python file? Does it have nay advantages?

jsilhan commented:
with this PR, the bash_completion plugin should be removed from core plugins. Then it will fix:
* https://bugzilla.redhat.com/show_bug.cgi?id=1207973
* https://bugzilla.redhat.com/show_bug.cgi?id=1201832

ignatenkobrain commented:
@pnemade, can you test this?

jsilhan commented:
Igor made a super cool completion features I would like you to be the first to test it.

pnemade commented:
1) one question here, when I manually copied files to only py3 lib dir, dnf completion not worked whereas on rawhide dnf is linked to dnf-3. I then copied completion_helper to py2 lib dir and it worked fine. Shouldn't just copying to py3 lib dir work here?

2) dnf completion is working and showing all the commands when you press TABTAB
3) But, I also see its partial results e.g. "dnf clean TABTAB" not showing sub-commands.
4) Then, "dnf help TABTAB" not showing now all the commands for completion
5) "yum history TABTAB" shows sub-commands whereas "dnf history TABTAB" not. 
6) "yum list TABTAB" shows sub-commands whereas "dnf list TABTAB" not.

Above 5,6 points are not implemented in current dnf releases also. So they are new RFE here.
If something is working for anyone from above list please ask me again to test.
Thanks.

ignatenkobrain commented:
> 1) one question here, when I manually copied files to only py3 lib dir, dnf completion not worked whereas on rawhide dnf is linked to dnf-3. I then copied completion_helper to py2 lib dir and it worked fine. Shouldn't just copying to py3 lib dir work here?

I've added more better detection.

> 2) dnf completion is working and showing all the commands when you press TABTAB

great

> 3) But, I also see its partial results e.g. "dnf clean TABTAB" not showing sub-commands.

fixed

> 4) Then, "dnf help TABTAB" not showing now all the commands for completion

fixed

> 5) "yum history TABTAB" shows sub-commands whereas "dnf history TABTAB" not.

fixed

> 6) "yum list TABTAB" shows sub-commands whereas "dnf list TABTAB" not.

fixed

pnemade commented:
:+1:  I confirm all the issues reported by me above are fixed by @ignatenkobrain's recent commits in this PR.

Thanks @ignatenkobrain for these fixes. I am really happy now to use these new dnf bash completions.

jsilhan commented:
Merged, thanks for cooperation, guys
